src/main/java/dan200/computercraft/ComputerCraft.java

```diff
@@ -644,6 +646,14 @@ public static void registerMediaProvider( IMediaProvider provider )
         }
     }
 
+    public static void registerAPIFactory( ILuaAPIFactory provider )
```

Comment:
Perhaps this should return a `boolean` specifying whether the API was actually registered?

Reply:
I'm only really doing it this way because that's what all the other methods do. The only way it could thrown an exception is if the reflection fails, which shouldn't ever happen anyway.

Reply:
Oh, I was referring to `if( provider != null && !apiFactories.contains( provider ) )` failing.